src/sqlparser/src/ast/utils.rs

```diff
@@ -0,0 +1,574 @@
+    }
+
+    fn serialize_seq(self, _len: Option<usize>) -> Result<Self::SerializeSeq, Self::Error> {
+        Err(Error::NotSupported("serialize_seq".into()))
+    }
+
+    fn serialize_tuple(self, _len: usize) -> Result<Self::SerializeTuple, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple".into()))
+    }
+
+    fn serialize_tuple_struct(
+        self,
+        _name: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeTupleStruct, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple_struct".into()))
+    }
+
+    fn serialize_tuple_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeTupleVariant, Self::Error> {
+        Err(Error::NotSupported("serialize_tuple_variant".into()))
+    }
+
+    fn serialize_map(self, _len: Option<usize>) -> Result<Self::SerializeMap, Self::Error> {
+        Ok(self)
+    }
+
+    fn serialize_struct(
+        self,
+        _name: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeStruct, Self::Error> {
+        Ok(self)
+    }
+
+    fn serialize_struct_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        _variant: &'static str,
+        _len: usize,
+    ) -> Result<Self::SerializeStructVariant, Self::Error> {
+        Err(Error::NotSupported("serialize_struct_variant".into()))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::Serialize;
+
+    use crate::ast::utils::SqlOptionVecSerializer;
+    use crate::ast::{Ident, ObjectName, SqlOption, Value};
+
+    fn to_object_name(s: &str) -> ObjectName {
+        ObjectName(vec![Ident::new_unchecked(s)])
+    }
+
+    #[test]
+    fn test_serializer_basic() {
+        #[derive(Serialize)]
+        struct Foo {
+            a: String,
+            #[serde(rename = "foo.2.b")]
+            b: u32,
+            c: Option<f32>,
+            d: Option<f32>,
+        }
+
+        let mut serializer = SqlOptionVecSerializer::default();
+        let foo = Foo {
+            a: "v_a".to_string(),
+            b: 2,
+            c: Some(1.5f32),
+            d: None,
+        };
+        foo.serialize(&mut serializer).unwrap();
+        let sql_option: Vec<SqlOption> = serializer.into();
+        assert_eq!(
+            sql_option,
+            vec![
+                SqlOption {
+                    name: to_object_name("a"),
+                    value: Value::SingleQuotedString("v_a".into())
+                },
+                SqlOption {
+                    name: to_object_name("foo.2.b"),
+                    value: Value::Number("2".into())
+                },
+                SqlOption {
+                    name: to_object_name("c"),
+                    value: Value::Number("1.5".into())
+                },
+                SqlOption {
+                    name: to_object_name("d"),
+                    value: Value::Null
+                },
```

Comment:
Should we omit the field that's unset? I guess we need an end-to-end test like 

```
SQL 
--parse--> AST
--deser--> Connector Options
--redact--> Redacted Connector Options
--ser--> AST
--unparse--> SQL
```

to check whether it works well.

Reply:
Omitted the field that's unset.
Modified `pubsub.slt` to cover the code path.